ethcore/src/lib.rs

```diff
@@ -16,23 +16,6 @@
 
 #![warn(missing_docs)]
 #![cfg_attr(feature="benches", feature(test))]
-#![cfg_attr(feature="dev", feature(plugin))]
-#![cfg_attr(feature="dev", plugin(clippy))]
-
-// Clippy settings
-// Most of the time much more readable
-#![cfg_attr(feature="dev", allow(needless_range_loop))]
```

Comment:
if we want to run clippy as a plugin, we should not remove those, but only rename to:

```
#[cfg_attr(feature="cargo-clippy", allow(needless_range_loop))]
```

Reply:
I tried running `cargo-clipp` as a plugin but it didn't work (I think parity failed to compile on nightly). Let's look at this from a fresh perspective when it works, some of the things here might not be valid any more, I would prefer to fix the code at some point rather than disable lints.
What do you think?

Reply:
fair enough ;)